src/shredder/reed_solomon.rs

```diff
@@ -218,10 +221,14 @@ mod tests {
     fn take_and_map_enough_shreds(
         data: Vec<DataShred>,
         coding: Vec<CodingShred>,
-    ) -> Vec<ValidatedShred> {
+    ) -> [Option<ValidatedShred>; TOTAL_SHREDS] {
         let sk = SecretKey::new(&mut rand::rng());
         let shreds = data_and_coding_to_output_shreds(data, coding, &sk);
         // reverse order to get coding shreds, not just data shreds
-        shreds.into_iter().rev().take(DATA_SHREDS).collect()
+        let mut shreds = shreds.into_iter().rev().map(Some).collect::<Vec<_>>();
+        for shred in shreds.iter_mut().skip(DATA_SHREDS) {
```

Comment:
unnecessary to destroy the order of the array here: just `take(TOTAL_SHREDS - DATA_SHREDS)` for the loop and remove `rev()` above

Reply:
also do we need to collect into a `Vec` before turning into an array? (ah, this is only for testing again? then it doesn't matter as much)